src/plugins/guided_onboarding/server/routes/index.ts

```diff
@@ -7,92 +7,154 @@
  */
 
 import { schema } from '@kbn/config-schema';
-import { IRouter, SavedObjectsClient } from '@kbn/core/server';
-import {
-  guidedSetupDefaultState,
-  guidedSetupSavedObjectsId,
-  guidedSetupSavedObjectsType,
-} from '../saved_objects';
-
-const doesGuidedSetupExist = async (savedObjectsClient: SavedObjectsClient): Promise<boolean> => {
-  return savedObjectsClient
-    .find({ type: guidedSetupSavedObjectsType })
-    .then((foundSavedObjects) => foundSavedObjects.total > 0);
+import type { IRouter, SavedObjectsClient } from '@kbn/core/server';
```

Comment:
I think it'd be good to add some API integration tests for the routes, but ran out of time this week. Opened https://github.com/elastic/kibana/issues/142261 as a follow-up task.